### ISSUE TITLE
Gracefully handle Youtube DASH-Segements [was: YouTube 404 error]

### ISSUE THREAD

**Issue by JamesTheAwesomeDude** [mpv-player/mpv#3388]
### mpv version and platform

mpv 0.18.0 on Linux Mint 17.3
Git latest YouTube-DL
### Reproduction steps

```
mpv ytdl://P_gg6fGppwM
```
### Expected behavior

Video plays
### Actual behavior

```
[ffmpeg] https: HTTP error 404 Not Found
Failed to open https://r5---sn-5uaeznd7.googlevideo.com/videoplayback/id/3ff820e9f1a9a703/itag/137/source/yt_otf/requiressl/yes/pl/19/mn/sn-5uaeznd7/mm/31/initcwndbps/1031250/ms/au/mv/m/nh/IgpwcjAxLmF0bDAxKgw3Mi4xNC4yMjAuMTY/ratebypass/yes/mime/video%2Fmp4/otfp/1/otf/1/lmt/1462234017178236/sver/3/mt/1470168243/upn/pRSV_sqqVh0/key/dg_yt0/signature/817B0E158E2F1C1D8EA8FCF9F405D9A79569A5F5.42947676DDD853F750C7452F417FCB44800AE972/fexp/9406015,9416891,9419451,9422596,9427769,9428398,9431012,9433096,9433221,9433946,9435526,9435876,9437066,9437553,9438312,9438327,9438478,9438526,9438662,9438805,9438822,9439210,9439358,9439581,9439652,9439688,9439812,9440376,9440880,9441414,9442425,9443264/ip/68.117.190.30/ipbits/0/expire/1470190479/sparams/ip,ipbits,expire,id,itag,source,requiressl,pl,mn,mm,initcwndbps,ms,mv,nh,ratebypass,mime,otfp,otf,lmt/.
```
### Log files

`mpv ytdl://P_gg6fGppwM -v`: http://sprunge.us/hOTA  
`youtube-dl P_gg6fGppwM -v`: http://sprunge.us/QeIX

**Comment by wiiaboo** [comment]
Dash segments. It's been *fixed* in https://github.com/mpv-player/mpv/commit/5bcb22beea14bb48a84c610425d5693e0305e434 already.

Until FFmpeg supports .mpd demuxing, add `[protocol!=http_dash_segments]` to your ytdl-format in mpv.

**Comment by ChrisK2** [comment]
> Until FFmpeg supports .mpd demuxing, add [protocol!=http_dash_segments] to your ytdl-format in mpv.

We should just do this as default format for now.

**Comment by wiiaboo** [comment]
https://github.com/rg3/youtube-dl/pull/10554

**Comment by mc4man** [comment]
Url works fine now here with updated youtube-dl

**Comment by wiiaboo** [comment]
Another sample with 17 min: https://www.youtube.com/watch?v=aclExbIzafg

**Comment by JamesTheAwesomeDude** [comment]
@wiiaboo those links work fine on my machine. YouTube-DL version 2017.01.31 (git `d7e215b`); MPV version `git-39adaf3`

````
[cplayer] Playing: ytdl://aclExbIzafg
[cplayer] Running hook: ytdl_hook/on_load
[global] config path: 'youtube-dl' -/-> '/home/james/.config/mpv/youtube-dl'
[global] config path: 'youtube-dl' -/-> '/home/james/.mpv/youtube-dl'
[global] config path: 'youtube-dl' -/-> '/etc/mpv/youtube-dl'
[osd/libass] Using font provider fontconfig
[osd/libass] Done.
[ytdl_hook] youtube-dl succeeded! 
[cplayer] Set property: file-local-options/user-agent="Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20150101 Firefox/47.0 (Chrome)" -> 1
[cplayer] Set property: stream-open-filename="https://r1---sn-5uaeznld.googlevideo.com/videoplayback?dur=1034.286&ratebypass=yes&usequic=no&expire=1485900930&nh=IgpwcjAyLmF0bDIyKgkxMjcuMC4wLjE&lmt=1457513525742874&sparams=dur%2Cei%2Cid%2Cinitcwndbps%2Cip%2Cipbits%2Citag%2Clmt%2Cmime%2Cmm%2Cmn%2Cms%2Cmv%2Cnh%2Cpl%2Cratebypass%2Crequiressl%2Csource%2Cupn%2Cusequic%2Cexpire&ipbits=0&requiressl=yes&initcwndbps=1256250&source=youtube&ei=IriQWN3dGZKYugKPtqcQ&mn=sn-5uaeznld&ip=68.117.190.30&mm=31&itag=22&pl=19&signature=7AA959B055C77B767503CBD0ACD7789EF0586775.6DFB0548954B78AC25885CEE2AA5C17A6B5A60E6&id=o-APZOFWMEfl1vw5P5fxxWtat5xg4zYwP-HEq__LeFed-z&mv=m&upn=H3702xRd3G4&mt=1485879025&ms=au&key=yt6&mime=video%2Fmp4" -> 1
[cplayer] Set property: file-local-options/force-media-title="EP 15: Welcome Back! Rambles and new pc" -> 1
[cplayer] Run command: hook-ack, flags=0, args=[on_load]
[ffmpeg] Opening https://r1---sn-5uaeznld.googlevideo.com/videoplayback?dur=1034.286&ratebypass=yes&usequic=no&expire=1485900930&nh=IgpwcjAyLmF0bDIyKgkxMjcuMC4wLjE&lmt=1457513525742874&sparams=dur%2Cei%2Cid%2Cinitcwndbps%2Cip%2Cipbits%2Citag%2Clmt%2Cmime%2Cmm%2Cmn%2Cms%2Cmv%2Cnh%2Cpl%2Cratebypass%2Crequiressl%2Csource%2Cupn%2Cusequic%2Cexpire&ipbits=0&requiressl=yes&initcwndbps=1256250&source=youtube&ei=IriQWN3dGZKYugKPtqcQ&mn=sn-5uaeznld&ip=68.117.190.30&mm=31&itag=22&pl=19&signature=7AA959B055C77B767503CBD0ACD7789EF0586775.6DFB0548954B78AC25885CEE2AA5C17A6B5A60E6&id=o-APZOFWMEfl1vw5P5fxxWtat5xg4zYwP-HEq__LeFed-z&mv=m&upn=H3702xRd3G4&mt=1485879025&ms=au&key=yt6&mime=video%2Fmp4
[ffmpeg] Mime-type: 'video/mp4'
[ffmpeg] Stream opened successfully.
[cache] Cache size set to 150000 KiB (75000 KiB backbuffer)
[demux] Trying demuxers for level=normal.
[lavf] Found 'mov,mp4,m4a,3gp,3g2,mj2' at score=100 size=2048.
[ffmpeg/video] h264: Reinit context to 1280x720, pix_fmt: yuv420p
[lavf] avformat_find_stream_info() finished after 467349 bytes.
[demux] Detected file format: mov,mp4,m4a,3gp,3g2,mj2 (libavformat)
[cache] blocking for STREAM_CTRL 6
[cplayer] Opening done: https://r1---sn-5uaeznld.googlevideo.com/videoplayback?dur=1034.286&ratebypass=yes&usequic=no&expire=1485900930&nh=IgpwcjAyLmF0bDIyKgkxMjcuMC4wLjE&lmt=1457513525742874&sparams=dur%2Cei%2Cid%2Cinitcwndbps%2Cip%2Cipbits%2Citag%2Clmt%2Cmime%2Cmm%2Cmn%2Cms%2Cmv%2Cnh%2Cpl%2Cratebypass%2Crequiressl%2Csource%2Cupn%2Cusequic%2Cexpire&ipbits=0&requiressl=yes&initcwndbps=1256250&source=youtube&ei=IriQWN3dGZKYugKPtqcQ&mn=sn-5uaeznld&ip=68.117.190.30&mm=31&itag=22&pl=19&signature=7AA959B055C77B767503CBD0ACD7789EF0586775.6DFB0548954B78AC25885CEE2AA5C17A6B5A60E6&id=o-APZOFWMEfl1vw5P5fxxWtat5xg4zYwP-HEq__LeFed-z&mv=m&upn=H3702xRd3G4&mt=1485879025&ms=au&key=yt6&mime=video%2Fmp4
[global] config path: 'sub/' -/-> '/home/james/.config/mpv/sub/'
[global] config path: 'sub/' -/-> '/home/james/.mpv/sub/'
[global] config path: 'sub/' -/-> '/etc/mpv/sub/'
[global] config path: 'audio/' -/-> '/home/james/.config/mpv/audio/'
[global] config path: 'audio/' -/-> '/home/james/.mpv/audio/'
[global] config path: 'audio/' -/-> '/etc/mpv/audio/'
[cplayer] Running hook: ytdl_hook/on_preloaded
[cplayer] Run command: hook-ack, flags=0, args=[on_preloaded]
[cplayer]  (+) Video --vid=1 (*) (h264)
[cplayer]  (+) Audio --aid=1 --alang=eng (*) (aac)
````

**Comment by wiiaboo** [comment]
Well yeah, we specifically blacklist dash segments right now, so it should work on git master. These are samples to test a wip branch.

**Comment by wiiaboo** [comment]
Fixed/implemented in 61202bb3640740d2cb98cf13922dcdf67970d5ef.
